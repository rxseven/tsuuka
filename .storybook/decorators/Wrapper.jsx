import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  padding: 2rem;
`;

export default Wrapper;
