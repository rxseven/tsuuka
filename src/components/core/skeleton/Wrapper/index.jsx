import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';
import styled from 'styled-components';

import HTML from 'constants/elements/html';

const Frame = styled.div.attrs({ id: HTML.wrapper })`
  display: flex;
  flex-direction: column;
  height: 100%;
`;

const propTypes = exact({
  children: PropTypes.node.isRequired
});

function Wrapper({ children }) {
  return <Frame>{children}</Frame>;
}

Wrapper.propTypes = propTypes;

export default Wrapper;
