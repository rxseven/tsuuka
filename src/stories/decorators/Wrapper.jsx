import React from 'react';
import styled from 'styled-components';

const Frame = styled.div`
  margin: -2rem;
`;

function Wrapper({ children }) {
  return <Frame>{children}</Frame>;
}

export default Wrapper;
