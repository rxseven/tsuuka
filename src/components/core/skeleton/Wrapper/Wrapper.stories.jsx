import React from 'react';
import { storiesOf } from '@storybook/react';

import mock from 'stories/mock';
import notes from 'stories/shared/notes';
import Wrapper from './index';

const { text } = mock;
const title = 'Components/Core/Skeleton/Wrapper';
const component = <Wrapper>{text}</Wrapper>;

storiesOf(title, module).add('default', () => component, {
  notes: notes.minimal
});
