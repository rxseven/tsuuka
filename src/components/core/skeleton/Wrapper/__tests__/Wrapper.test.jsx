import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Wrapper from '../index';

describe('<Wrapper />', () => {
  // Arrange
  const props = { children: 'content' };
  const component = <Wrapper {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
