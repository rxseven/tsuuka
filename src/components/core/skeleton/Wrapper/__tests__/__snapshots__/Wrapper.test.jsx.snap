// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> Snapshot tests should render correctly 1`] = `
<div
  className="sc-bdVaJa jaytjN"
>
  content
</div>
`;
